smithy-smoke-test-traits/src/main/resources/META-INF/smithy/smithy.test.smithy

```diff
@@ -0,0 +1,91 @@
+$version: "2.0"
+
+namespace smithy.test
+
+/// Defines a set of test cases to send to a live service to ensure that a
+/// client can successfully connect to a service and get the right kind of
+/// response.
+@trait(selector: "operation")
+list smokeTests {
+    /// A list of smoke tests to perform on the operation
+    member: SmokeTestCase
+}
+
+/// A single smoke test case definition.
+@private
+structure SmokeTestCase {
```

Comment:
I feel like we should share some test case definitions, maybe as mixins. A lot of this is exactly the same as protocol testsvand endpoint tests. It might make writing generators easier too if we have shared interfaces.

Reply:
Yea, there's a lot of crossover, but I wanted to just focus on smoke tests in this PR. If you think we should try to get it all in one though I can update it.